client/src/App.js

```diff
@@ -1,15 +1,67 @@
+import { Switch, Route } from 'react-router-dom';
+import styled from 'styled-components';
 import Home from './Pages/Home';
+import Watchlist from './Pages/Watchlist';
+import Friends from './Pages/Friends';
+import Search from './Pages/Search';
 import Banner from '../src/components/Banner';
-import styled from 'styled-components';
+import Navigation from '../src/components/Navigation';
+import Store from './hooks/Store';
+import { useLocalStorage } from '../src/hooks/useLocalStorage';
 
 function App() {
+  const [watchlist, setWatchList] = useLocalStorage('Watchlist', []);
+
+  const isOnWatchList = (movieToAdd) =>
+    watchlist.some((movie) => movie.id === movieToAdd.id);
+
+  function addToWatchList(movieToAdd) {
+    if (!isOnWatchList(movieToAdd)) {
+      setWatchList([...watchlist, movieToAdd]);
+    } else
+      setWatchList(watchlist.filter((movie) => movie.id !== movieToAdd.id));
```

Comment:
I'm not sure if I read your code correctly but do you really need the "else"? If the movieToAdd is already on your wishlist simply doing "nothing" could maybe also be an option ;)

Reply:
The button handles "add" and "remove" at the same time, depending on whether the movie is on the watchlist or not. The "else" line handles the removal. That is why I also have a ternary on the button, which switches from "add to" to "remove from" Watchlist .So 1 button for two functionalities.  ;)

Reply:
Ahh okay, that makes sense - need to clone the repository next time ;).